app/models/balance_sheet.rb

```diff
@@ -59,6 +59,17 @@ def account_groups(classification = nil)
           account.define_singleton_method(:weight) do
             classification_total.zero? ? 0 : account.converted_balance / classification_total.to_d * 100
           end
+
+          # Define our subtype labeling logic with short format (for sidebar)
+          account.define_singleton_method(:short_subtype_label) do
+            accountable.short_subtype_label_for(account.subtype) || accountable.display_name
+          end
+
+          # Define long label method (for other views that might need it)
+          account.define_singleton_method(:long_subtype_label) do
+            accountable.long_subtype_label_for(account.subtype) || accountable.display_name
+          end
```

Comment:
If I'm not mistaken, now that you've got both of these defined on `account.rb` directly, we can remove these singleton overrides here and things will work fine in the sidebar.